src/Jackett.Common/Utils/Extensions.cs

```diff
@@ -85,29 +85,24 @@ public static class KeyValuePairsExtension
 
     public static class ParseExtension
     {
-        public static T? TryParse<T>(this string value) where T : struct
+        public static T? TryParse<T>(this string value) where T : struct, IConvertible
         {
-            var type = typeof(T);
-            var parseMethods = type.GetMethods(System.Reflection.BindingFlags.Public | System.Reflection.BindingFlags.Static).Where(m => m.Name == "Parse");
-            var parseMethod = parseMethods.Where(m =>
-            {
-                var parameters = m.GetParameters();
-                var hasOnlyOneParameter = parameters.Count() == 1;
-                var firstParameterIsString = parameters.First().ParameterType == typeof(string);
-
-                return hasOnlyOneParameter && firstParameterIsString;
-            }).First();
-            if (parseMethod == null)
+            if (string.IsNullOrWhiteSpace(value))
                 return null;
             try
             {
-                var val = parseMethod.Invoke(null, new object[] { value });
+                var val = Convert.ChangeType(value, typeof(T));
                 return (T)val;
             }
-            catch
+            catch(Exception e) when
+                (e is InvalidCastException // Conversion not supported
+                 || e is FormatException) // Conversion failed
```

Comment:
Just catch ALL exceptions and return null. This method is not expected to fail.

Reply:
Empty catch or `catch(Exception)` are both generally considered very bad practice. At a minimum we should not be catching `ArgumentNullException` because `typeof(T)` should never return null and something has gone horribly wrong in ways that shouldn't be hidden if it does.

Should an OverflowException return null? Or should it return MaxValue or maybe a modulus of MaxValue (wrapped value)?

It is obvious that not being able to convert a string to the target value should default to null, and it is obvious that if the string is not a valid target value that it should also return null. The other two above do not have an obvious handling path, IMHO.

Reply:
As general rule, the methods called TrySomething() never throw an exception. They will do his best to do the job or fail in silence. Could be return null or converted value + boolean like in .Net API.